src/OpenTelemetry.AutoInstrumentation/Loading/Initializers/AspNetCoreInitializer.cs

```diff
@@ -22,12 +28,33 @@ public override void Initialize(ILifespanManager lifespanManager)
         var httpInListenerType = Type.GetType("OpenTelemetry.Instrumentation.AspNetCore.Implementation.HttpInListener, OpenTelemetry.Instrumentation.AspNetCore")!;
 
         var options = new OpenTelemetry.Instrumentation.AspNetCore.AspNetCoreTraceInstrumentationOptions();
+
+        if (_tracerSettings.InstrumentationOptions.AspNetCoreInstrumentationCaptureRequestHeaders.Count != 0)
+        {
+            options.EnrichWithHttpRequest = EnrichWithHttpRequest;
```

Comment:
for plugins we need to create dynamic call that wraps both, the current value and our function call.

Reply:
maybe it's that simple > https://stackoverflow.com/questions/40401610/possible-to-combine-two-actions-in-c

Reply:
> For plugins we need to create dynamic call that wraps both, the current value and our function call. Also need to make sure that plugins assign the value first.

Current implementation - we are setting what we need, then sending it to the plugin. I think that it should stay as is. Plugin potentially would like to remove the feature. I agree that we should document this change (in changelog and in plugin.md).

Reply:
On the other hand, plugin can handle this also when documented. We definitely use it already, so it's important for us also.

Reply:
Please check a15789f722179866b935d6d1dcb6acb225f34d10